### ISSUE TITLE
Converting a RichTextField to StreamField

### ISSUE THREAD

**Issue by Squarehost** [wagtail/wagtail#4395]
Hi I recently started using wagtail and am enjoying it, I much prefer it to DjangoCMS, but hopefully someone can help me out with this.

### Issue Summary

Been trying to convert a RichTextField to a StreamField but getting errors following the example given in the docs http://docs.wagtail.io/en/v2.0/topics/streamfield.html#streamfield-migrating-richtext

This is the Traceback that I get:

```
$ python manage.py migrate blog
System check identified some issues:

Operations to perform:
  Apply all migrations: blog
Running migrations:
  Applying blog.0008_auto_20180316_1650...BODY <class 'str'> <div class="rich-text"><p>[vc_row][vc_column][vc_column_text]<span style="color: #555555;">[real3dflipbook id="1"]</span>[/vc_column_text][/vc_column][/vc_row]</p></div>
Traceback (most recent call last):
  File "manage.py", line 12, in <module>
    execute_from_command_line(sys.argv)
  File "site-packages/django/core/management/__init__.py", line 371, in execute_from_command_line
    utility.execute()
  File "site-packages/django/core/management/__init__.py", line 365, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "site-packages/django/core/management/base.py", line 288, in run_from_argv
    self.execute(*args, **cmd_options)
  File "site-packages/django/core/management/base.py", line 335, in execute
    output = self.handle(*args, **options)
  File "site-packages/django/core/management/commands/migrate.py", line 200, in handle
    fake_initial=fake_initial,
  File "site-packages/django/db/migrations/executor.py", line 117, in migrate
    state = self._migrate_all_forwards(state, plan, full_plan, fake=fake, fake_initial=fake_initial)
  File "site-packages/django/db/migrations/executor.py", line 147, in _migrate_all_forwards
    state = self.apply_migration(state, migration, fake=fake, fake_initial=fake_initial)
  File "site-packages/django/db/migrations/executor.py", line 244, in apply_migration
    state = migration.apply(state, schema_editor)
  File "site-packages/django/db/migrations/migration.py", line 122, in apply
    operation.database_forwards(self.app_label, schema_editor, old_state, project_state)
  File "site-packages/django/db/migrations/operations/special.py", line 190, in database_forwards
    self.code(from_state.apps, schema_editor)
  File "site-packages/blog/migrations/0008_auto_20180316_1650.py", line 11, in convert_to_streamfield
    if page.body.raw_text and not page.body:
AttributeError: 'str' object has no attribute 'raw_text'
```

My Model (Im actually using https://github.com/SquarehostLtd/wagtail_blog/blob/wagtail-2.0-support/) but I'll only show the relevant field that i'm migrating
```
class BlogPage(Page):
    body = StreamField([
        ('rich_text', blocks.RichTextBlock()),
    ])
    # body = RichTextField(verbose_name=_('body'), blank=True)
```

and here is the migration file
```
# Generated by Django 2.0.2 on 2018-03-16 16:50

from django.db import migrations
from wagtail.core.rich_text import RichText


def convert_to_streamfield(apps, schema_editor):
    BlogPage = apps.get_model("blog", "BlogPage")
    for page in BlogPage.objects.all():
        if page.body.raw_text and not page.body:
            page.body = [('rich_text', RichText(page.body.raw_text))]
            page.save()


def convert_to_richtext(apps, schema_editor):
    BlogPage = apps.get_model("blog", "BlogPage")
    for page in BlogPage.objects.all():
        if page.body.raw_text is None:
            raw_text = ''.join([
                child.value.source for child in page.body
                if child.block_type == 'rich_text'
            ])
            page.body = raw_text
            page.save()

class Migration(migrations.Migration):

    dependencies = [
        ('blog', '0007_auto_20180220_1115'),
    ]

    operations = [
        migrations.RunPython(
            convert_to_streamfield,
            convert_to_richtext,
        ),
    ]

```

and an adaptation I've tried due to the error above no attribute raw_text
```
# Generated by Django 2.0.2 on 2018-03-16 16:50

from django.db import migrations
from wagtail.core.rich_text import RichText
from wagtail.core.blocks import StreamValue


def convert_to_streamfield(apps, schema_editor):
    BlogPage = apps.get_model("blog", "BlogPage")
    for page in BlogPage.objects.all():
        print('BODY', type(page.body), RichText(page.body))
        if page.body:
            page.body = [('rich_text', RichText(page.body))]
            page.save()
            print('PAGE_SAVED', page.body)


def convert_to_richtext(apps, schema_editor):
    BlogPage = apps.get_model("blog", "BlogPage")
    print('TEST123')
    for page in BlogPage.objects.all():
        if page.body.raw_text is None:
            raw_text = ''.join([
                child.value.source for child in page.body
                if child.block_type == 'rich_text'
            ])
            page.body = raw_text
            page.save()

class Migration(migrations.Migration):

    dependencies = [
        ('blog', '0007_auto_20180220_1115'),
    ]

    operations = [
        migrations.RunPython(
            convert_to_streamfield,
            convert_to_richtext,
        ),
    ]

```

With my changed migration no errors occur but when going to the admin I don't see any content just the chooser box.

Looking at the mode instance in the shell this is what I see:

```
>>> from blog.models import BlogPage
>>> bp = BlogPage.objects.first()
>>> bp.body
[]
>>> bp.body.raw_text
"[('rich_text', <wagtail.core.rich_text.RichText object at 0x1115d3518>)]"
```

using Python 3.6.4, Django 2.0.3, Wagtail 2.0

Hopefully this is something simple and I'm just missing something, I look forward to getting this problem solved.

Thanks.

**Comment by IlyaSemenov** [comment]
You need to start with `makemigrations` without `--empty`, and leave the auto generated AlterTable migration in place:

```python
	operations = [
		migrations.AlterField(
			model_name='homepage',
			name='body',
			field=wagtail.core.fields.StreamField(...),
		),
		migrations.RunPython(
			convert_to_streamfield,
			convert_to_richtext,
		),
	]
```

**Comment by cssidy** [comment]
I also get this error when following the Wagtail docs. Same set up and same error logs. When I leave in the AlterField as @IlyaSemenov suggests, the migration passes. But then upon reloading my app the new StreamField UI is unusable. None of the previous rich text was converted over, it has totally disappeared. And when I click an icon in the StreamField box nothing happens. When I click the plus-shaped open/close button on the StreamField it gets stuck and unable to open again.

**Comment by gasman** [comment]
Documentation fixed in #5207.
